### DISCUSSION TITLE
Windows support

### DISCUSSION THREAD

**Discussion by lovepeacemz** [seart-group/java-tree-sitter#107] [Q&A]
Theoretically, I want to know if it support Windows?

**Comment by dabico** [comment]
Hi @lovepeacemz

I've migrated your question to the discussions since it's a more fitting section for this inquiry type. Unfortunately, this binding does not support Windows at the moment. To build it, we use the [Makefile](https://raw.githubusercontent.com/tree-sitter/tree-sitter/master/Makefile) from tree-sitter, which does not seem to support Windows. I have noticed that their releases also include Windows distributions, but I do not know how they build these libraries at the moment. Maybe looking into their CI could yield some insight. Until we figure out how it should be done and how it can be best integrated within our existing build, Windows will not be supported.

**Reply by lovepeacemz** [reply]
“ I have noticed that their releases also include Windows distributions” means [serenadeai](https://github.com/serenadeai).?,

**Reply by dabico** [reply]
No, I mean the official [tree-sitter](https://github.com/tree-sitter/tree-sitter).

**Reply by lovepeacemz** [reply]
ok, In fact, I really want this feature in one or two month later, it there the plan that support it? and by the way, https://github.com/serenadeai/java-tree-sitter/issues/6 , may be this can help you?

**Reply by dabico** [reply]
> ok, In fact, I really want this feature in one or two month later, it there the plan that support it? 

It's not a priority for us to introduce this feature at this time.

> and by the way, [serenadeai/java-tree-sitter#6](https://github.com/serenadeai/java-tree-sitter/issues/6) , may be this can help you?

This won't help me because the compilation done by that user was on the other binding, which has a drastically different JNI implementation and overall library API. It won't work within the context of this project.
